File: clients/java/src/main/java/com/svix/coyote/apis/Kv.java
// this file is @generated
package com.svix.coyote.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.coyote.ApiException;
import com.svix.coyote.HttpClient;
import com.svix.coyote.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.coyote.models.KvDeleteIn;
import com.svix.coyote.models.KvDeleteOut;
import com.svix.coyote.models.KvGetIn;
import com.svix.coyote.models.KvGetNamespaceIn;
import com.svix.coyote.models.KvGetNamespaceOut;
import com.svix.coyote.models.KvGetOut;
import com.svix.coyote.models.KvSetIn;
import com.svix.coyote.models.KvSetOut;

public class Kv {
    private final HttpClient client;

    public Kv(HttpClient client) {
        this.client = client;
    }

    /** KV Set */
    public KvSetOut set(
        final KvSetIn kvSetIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/kv/set");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            kvSetIn,
            KvSetOut.class
            );
    }

    /** KV Get */
    public KvGetOut get(
        final KvGetIn kvGetIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/kv/get");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            kvGetIn,
            KvGetOut.class
            );
    }

    /** Get KV namespace */
    public KvGetNamespaceOut getNamespace(
        final KvGetNamespaceIn kvGetNamespaceIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/kv/get-namespace");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            kvGetNamespaceIn,
            KvGetNamespaceOut.class
            );
    }

    /** KV Delete */
    public KvDeleteOut delete(
        final KvDeleteIn kvDeleteIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/kv/delete");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            kvDeleteIn,
            KvDeleteOut.class
            );
    }
}

File: clients/python/diom/apis/kv.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    KvDeleteIn,
    KvDeleteOut,
    KvGetIn,
    KvGetNamespaceIn,
    KvGetNamespaceOut,
    KvGetOut,
    KvSetIn,
    KvSetOut,
)


class KvAsync(ApiBase):
    async def set(
        self,
        kv_set_in: KvSetIn,
    ) -> KvSetOut:
        """KV Set"""
        body = kv_set_in.model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/kv/set",
            body=body,
            response_type=KvSetOut,
        )

    async def get(
        self,
        kv_get_in: KvGetIn,
    ) -> KvGetOut:
        """KV Get"""
        body = kv_get_in.model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/kv/get",
            body=body,
            response_type=KvGetOut,
        )

    async def get_namespace(
        self,
        kv_get_namespace_in: KvGetNamespaceIn,
    ) -> KvGetNamespaceOut:
        """Get KV namespace"""
        body = kv_get_namespace_in.model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/kv/get-namespace",
            body=body,
            response_type=KvGetNamespaceOut,
        )

    async def delete(
        self,
        kv_delete_in: KvDeleteIn,
    ) -> KvDeleteOut:
        """KV Delete"""
        body = kv_delete_in.model_dump(exclude_unset=True, by_alias=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/kv/delete",
            body=body,
            response_type=KvDeleteOut,
        )


class Kv(ApiBase):
    def set(
        self,
        kv_set_in: KvSetIn,
    ) -> KvSetOut:
        """KV Set"""
        body = kv_set_in.model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/kv/set",
            body=body,
            response_type=KvSetOut,
        )

    def get(
        self,
        kv_get_in: KvGetIn,
    ) -> KvGetOut:
        """KV Get"""
        body = kv_get_in.model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/kv/get",
            body=body,
            response_type=KvGetOut,
        )

    def get_namespace(
        self,
        kv_get_namespace_in: KvGetNamespaceIn,
    ) -> KvGetNamespaceOut:
        """Get KV namespace"""
        body = kv_get_namespace_in.model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/kv/get-namespace",
            body=body,
            response_type=KvGetNamespaceOut,
        )

    def delete(
        self,
        kv_delete_in: KvDeleteIn,
    ) -> KvDeleteOut:
        """KV Delete"""
        body = kv_delete_in.model_dump(exclude_unset=True, by_alias=True)

        return self._request_sync(
            method="post",
            path="/api/v1/kv/delete",
            body=body,
            response_type=KvDeleteOut,
        )
